charts/consul/templates/gateway-resources-job.yaml

```diff
@@ -113,6 +113,34 @@ spec:
             - name: config
               mountPath: /consul/config
               readOnly: true
+        {{ if .Values.meshGateway.enabled }}
+        - name: mesh-gateway-resources
```

Comment:
So, all of this stuff should probably just be passed as conditional flags to the job in the above  container definition instead of making a secondary container definition. You may also want to adjust the flag names on the job accordingly since this is now for both mesh stuff and api gateway stuff, which I'd imagine means you may want to prefix any api gateway specific flags with something like `api-gateway`, so something like `-service-type` would be `-api-gateway-service-type`. Mesh gateway specific stuff could just be prefixed with `mesh-gateway` and then anything shared wouldn't have a prefix.